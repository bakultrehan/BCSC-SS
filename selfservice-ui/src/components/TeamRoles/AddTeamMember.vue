/** * Add team member of app */

<template>
  <v-container>
    <v-row class="text-left">
      <v-col v-if="userDetails">
        <v-card class="v-form ma-3" flat>
          <v-form ref="form" v-model="form">
            <v-card class="v-form px-6 ma-3">
              <v-alert type="error" dense outlined class="text-left" v-if="memberErrorStatus">
                <div>{{ $t('addTeamMember.errorTitle') }}</div>
                <ul>
                  <li
                    v-for="(errors, idx) in errorList"
                    :key="idx"
                    v-html="$t(`addTeamMember.${errors}`)"
                  ></li>
                </ul>
              </v-alert>

              <v-row>
                <v-col cols="12">
                  <div class="display-1">
                    {{
                    $t(
                    !editMode
                    ? 'addTeamMember.pagetitle'
                    : 'addTeamMember.pagetitleUpdate'
                    )
                    }}
                  </div>
                </v-col>
                <v-col cols="12" sm="6">
                  <v-card-subtitle class="headline bc-padding-left-0">
                    {{
                    $t('addTeamMember.contactInfo')
                    }}
                  </v-card-subtitle>
                  <Input
                    v-model="userDetails.firstName"
                    :label="$t('addTeamMember.labelFirstName')"
                    :rules="[
                      rules.required,
                      rules.length(1),
                      rules.maxLength(250)
                    ]"
                    type="text"
                    :disabled="disabled()"
                  />
                  <Input
                    v-model="userDetails.lastName"
                    :label="$t('addTeamMember.labelLastName')"
                    :rules="[
                      rules.required,
                      rules.length(1),
                      rules.maxLength(250)
                    ]"
                    type="text"
                    :disabled="disabled()"
                  />
                  <Input
                    v-model="userDetails.email"
                    :label="$t('addTeamMember.labelWorkEmail')"
                    :rules="[rules.required, rules.email, rules.maxLength(250)]"
                    type="text"
                    :disabled="disabled()"
                  />

                  <Input
                    v-model="userDetails.phone"
                    :label="$t('addTeamMember.labelPhone')"
                    type="text"
                    :disabled="disabled()"
                    :optional="true"
                  />
                </v-col>
                <v-col>
                  <v-divider class="mx-4" :inset="inset" vertical></v-divider>
                </v-col>
                <v-col cols="12" sm="5">
                  <v-card-subtitle class="headline bc-padding-left-0">
                    {{
                    $t('addTeamMember.roleTitle')
                    }}
                  </v-card-subtitle>

                  <v-radio-group v-model="userDetails.role" :mandatory="false">
                    <v-radio
                      :label="$t(`addTeamMember.labelRole${rolesList[1]}`)"
                      :value="projectRoles.developer"
                      class="my-2"
                    ></v-radio>
                    <v-radio
                      :label="$t(`addTeamMember.labelRole${rolesList[2]}`)"
                      :value="projectRoles.manager"
                      class="my-2"
                    ></v-radio>
                    <v-radio
                      :label="$t(`addTeamMember.labelRole${rolesList[3]}`)"
                      :value="projectRoles.cto"
                      class="my-2"
                    ></v-radio>
                  </v-radio-group>

                  <v-card-actions class="btn-bottom">
                    <v-spacer></v-spacer>
                    <Button
                      @click="$emit('toggleAddMember', false)"
                      aria-label="Back Button"
                      secondary
                    >{{ $t('addTeamMember.btnCancel') }}</Button>
                    <Button
                      :disabled="!form"
                      class="white--text submit-package ml-6"
                      @click="submitTeamMember"
                    >{{ $t('addTeamMember.btnSumbmit') }}</Button>
                  </v-card-actions>
                </v-col>
              </v-row>
            </v-card>
          </v-form>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
// import { ProjectUserModel, ProjectInfoModel } from '@/models/ProjectInfoModel';
import Input from '@/Atomic/Input/Input.vue';
import Button from '@/Atomic/Button/Button.vue';

import validationRules from '@/config/validationRules';
import { projectRoles, projectRolesList } from '@/constants/enums';
import { TeamRoleModel } from '@/models/TeamRoleModel';
import { memberDetails } from '@/store/modules/TeamRoles/defaults';

const TeamRolesModule = namespace('TeamRolesModule');
const KeyCloakModule = namespace('KeyCloakModule');

@Component({
  components: {
    Input,
    Button
  }
})
export default class AddTeamMember extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @Prop()
  public toggleAddMember: any;
  @Prop({ default: 0 })
  public memberId!: number;

  @TeamRolesModule.Getter('getTeamList')
  public teamList!: any;
  @TeamRolesModule.Action('addTeamMember')
  public addTeamMember!: any;
  @TeamRolesModule.Action('updateTeamMember')
  public updateTeamMember!: any;
  @TeamRolesModule.Action('getTeamMember')
  public getTeamMember!: any;
  @TeamRolesModule.Getter('memberErrorStatus')
  public memberErrorStatus!: any;
  @TeamRolesModule.Getter('getMemberErrorList')
  public getMemberErrorList!: any;
  @TeamRolesModule.Getter('getMemberDetails')
  public memberDetails!: any;
  @TeamRolesModule.Getter('memberSucessStatus')
  public memberSucessStatus!: any;
  @TeamRolesModule.Action('clearMemberData')
  public clearMemberData!: any;

  public errorList: any = {};

  public userDetails: TeamRoleModel;

  @KeyCloakModule.Getter('isAdmin')
  private isAdmin!: any;

  private rules: any = validationRules;
  private projectRoles: any = projectRoles;
  private selectedRole: any = 1;
  private rolesList: any = projectRolesList;
  private form: boolean = false;
  private editMode: boolean = false;

  public constructor() {
    super();
    this.userDetails = memberDetails;
  }

  @Watch('getMemberErrorList')
  private ongetMemberErrorListChanged(val: any) {
    this.errorList = val;
  }

  @Watch('memberDetails')
  private ongetmemberDetailsChanged(val: any) {
    this.userDetails = val;
  }

  @Watch('memberId')
  private onmemberIdChanged(val: any) {
    this.memberId = val;
    this.getMemberDetails();
  }
  @Watch('memberSucessStatus')
  private onmemberSucessStatusChanged(val: boolean) {
    if (val) {
      this.$emit('toggleAddMember', false);
    }
  }

  private submitTeamMember() {
    if (this.editMode) {
      this.updateTeamMember({
        userDetails: this.userDetails,
        projectId: this.id,
        memberId: this.memberId
      });
    } else {
      this.addTeamMember({
        userDetails: this.userDetails,
        projectId: this.id
      });
    }
  }
  private getMemberDetails() {
    if (this.memberId !== 0) {
      this.editMode = true;
      this.getTeamMember({ memberId: this.memberId, projectId: this.id });
    } else {
      this.editMode = false;
      this.clearMemberData();
    }
  }
  private disabled() {
    if (this.editMode && !this.isAdmin) {
      return true;
    }
    return false;
  }

  private mounted() {
    this.getMemberDetails();
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.v-text-field input {
  padding: 3px 0 3px 0;
}
.btn-bottom {
  bottom: 32px;
  position: absolute;
}
</style>
