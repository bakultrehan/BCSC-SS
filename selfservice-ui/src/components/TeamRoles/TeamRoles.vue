/** * Dashboard of app */

<template>
  <v-container>
    <v-row class="ma-5" v-if="isLoading">
      <v-col cols="12">
        <Loading />
      </v-col>
    </v-row>

    <v-row class="ma-5" v-else>
      <v-col cols="12">
        <v-card flat>
          <h2 class="text-left tab-headline page-title-h2">
            {{ $t('teamRoles.pagetitle') }}
          </h2>
          <p class="text-left  pageinfo">{{ $t('teamRoles.pageInfo') }}</p>
          <v-simple-table class="mt-5" v-if="teamList.length > 0">
            <template v-slot:default>
              <thead class="tbl-head">
                <tr class="text-left">
                  <th :scope="$t('teamRoles.tblHeadName')" class="text-left">
                    {{ $t('teamRoles.tblHeadName') }}
                  </th>
                  <th :scope="$t('teamRoles.tblHeadMyRole')" class="text-left">
                    {{ $t('teamRoles.tblHeadMyRole') }}
                  </th>
                  <th :scope="$t('teamRoles.tblHeadEmail')" class="text-left">
                    {{ $t('teamRoles.tblHeadEmail') }}
                  </th>
                  <th :scope="$t('teamRoles.tblHeadPhone')" class="text-left">
                    {{ $t('teamRoles.tblHeadPhone') }}
                  </th>
                  <th scope="actions" class="text-left" v-if="isAdmin"></th>
                </tr>
              </thead>
              <tbody>
                <tr v-for="team in teamList" :key="team.id" class="text-left">
                  <td>{{ team.firstName }} {{ team.lastName }}</td>
                  <td>
                    {{ $t(`teamRoles.labelRole${rolesList[team.role]}`) }}
                    <span
                      @click="
                        toggleAddMember(true, team.id, team.isCurrentUser)
                      "
                      @keyup.enter="
                        toggleAddMember(true, team.id, team.isCurrentUser)
                      "
                      v-if="team.isCurrentUser && teamList.length == 1"
                      class="edit-wrapper"
                      tabindex="0"
                      :aria-label="$t('global.edit')"
                      role="link"
                    >
                      <v-icon class="ml-2" small>mdi-pencil</v-icon>
                    </span>
                  </td>
                  <td>
                    <a :href="`mailto:${team.email}`">{{ team.email }}</a>
                  </td>
                  <td>{{ team.phone }}</td>
                  <td v-if="isAdmin">
                    <v-icon
                      @click="toggleAddMember(true, team.id)"
                      @keyup.enter="toggleAddMember(true, team.id)"
                      class="ml-2"
                      small
                      >mdi-pencil</v-icon
                    >
                    <v-icon
                      @click="deleteMemberDialog(team.id)"
                      @keyup.enter="deleteMemberDialog(true, team.id)"
                      class="ml-2 delete-member"
                      small
                      >mdi-delete</v-icon
                    >
                  </td>
                </tr>
              </tbody>
            </template>
          </v-simple-table>
          <v-card-actions class="mt-5">
            <v-spacer></v-spacer>
            <Button
              @click="toggleAddMember(true)"
              @keyup.enter="toggleAddMember(true)"
              :aria-label="$t('teamRoles.btnAddTeamMember')"
              :disabled="teamList.length >= 3"
              class="team-roles"
              data-test-id="btn-add-new-team"
              >{{ $t('teamRoles.btnAddTeamMember') }}</Button
            >
          </v-card-actions>
        </v-card>
      </v-col>

      <v-col>
        <div class="text-center">
          <v-dialog v-model="dialog" width="85%" class="text-left">
            <v-card>
              <AddTeamMember
                :id="id"
                @toggleAddMember="toggleAddMember"
                :memberId="memberId"
                :isCurrentUser="isCurrentUser"
              />
            </v-card>
          </v-dialog>
        </div>
      </v-col>
      <v-col>
        <v-dialog v-model="dialogDelete" persistent max-width="400">
          <v-card>
            <v-card-text
              class="text-left"
              v-html="$t('teamRoles.deleteWarning')"
            ></v-card-text>
            <v-card-actions>
              <v-spacer></v-spacer>
              <Button
                secondary
                text
                @click="deleteMember(true)"
                @keyup.enter="deleteMember(true)"
                >Cancel</Button
              >
              <Button
                text
                @click="deleteMember()"
                @keyup.enter="deleteMember(true)"
                >Delete</Button
              >
            </v-card-actions>
          </v-card>
        </v-dialog>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import { Component, Vue, Prop, Watch } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';

import Loading from '@/Atomic/Loading/Loading.vue';
import Button from '@/Atomic/Button/Button.vue';
import AddTeamMember from '@/components/TeamRoles/AddTeamMember.vue';

import { projectRoles } from '@/constants/enums';

const TeamRolesModule = namespace('TeamRolesModule');
const KeyCloakModule = namespace('KeyCloakModule');

@Component({
  components: {
    Loading,
    Button,
    AddTeamMember,
  },
})
export default class TeamRoles extends Vue {
  @Prop({ default: 0 })
  public id!: number;
  @TeamRolesModule.Getter('getTeamList')
  public teamList!: any;
  @TeamRolesModule.Action('loadTeam')
  public loadTeam!: any;
  @TeamRolesModule.Action('deleteTeamMember')
  public deleteTeamMember!: any;
  @KeyCloakModule.Getter('isAdmin')
  private isAdmin!: any;

  private isLoading: boolean = false;
  // private selectedTab: number = 0;
  private rolesList: any = projectRoles;
  private dialog: boolean = false;
  private dialogDelete: boolean = false;
  private memberId: number = 0;
  private isCurrentUser: boolean = false;

  @Watch('teamList')
  private ongetroleListChanged(val: any) {
    this.isLoading = false;
  }

  private toggleAddMember(
    status: boolean = false,
    memberId: number = 0,
    isCurrentUser: boolean = false
  ) {
    this.memberId = memberId;
    this.isCurrentUser = isCurrentUser;
    this.dialog = status;
  }

  private deleteMemberDialog(teamMemberId: number) {
    this.dialogDelete = true;
    this.memberId = teamMemberId;
  }

  private deleteMember(cancel: boolean = false) {
    if (!cancel) {
      this.isLoading = true;
      this.deleteTeamMember({
        projectId: this.id,
        memberId: this.memberId,
      });
    }
    this.memberId = 0;
    this.dialogDelete = false;
  }

  private mounted() {
    this.isLoading = true;
    this.loadTeam(this.id);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.tbl-head {
  background-color: $BCgovBlue10;
  color: $BCgovFontColorInverted !important;
  & th {
    color: $BCgovFontColorInverted !important;
  }
}
</style>
