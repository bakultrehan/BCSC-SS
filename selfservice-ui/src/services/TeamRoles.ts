import axios from '@/lib/axios';
// import { ProjectInfoModel } from '@/models/ProjectInfoModel';
import { TEAMROLE_URL } from '@/config/api-endpoints';
import { getUrl } from '@/lib/helpers';

export class TeamRoles {
  public static async getTeamRoles(projectId: number) {
    const teamURL = getUrl(TEAMROLE_URL, projectId);

    return await axios.get(teamURL);
  }
  public static async addTeamMember(userDetails: any, projectId: number) {
    const teamURL = getUrl(TEAMROLE_URL, projectId);

    return await axios.post(teamURL, userDetails);
  }
  public static async getTeamMember(projectId: number, memberId: number) {
    const teamURL = getUrl(TEAMROLE_URL, projectId);

    return await axios.get(`${teamURL}/${memberId}`);
  }
  public static async updateTeamMember(
    userDetails: any,
    projectId: number,
    memberId: number
  ) {
    const teamURL = getUrl(TEAMROLE_URL, projectId);

    return await axios.put(`${teamURL}/${memberId}`, userDetails);
  }
  public static async deleteTeamMember(projectId: number, memberId: number) {
    const teamURL = getUrl(TEAMROLE_URL, projectId);
    return await axios.delete(`${teamURL}/${memberId}`);
  }
}
