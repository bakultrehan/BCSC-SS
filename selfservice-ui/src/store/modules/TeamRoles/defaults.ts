const memberDetails = {
  email: '',
  phone: '',
  firstName: '',
  lastName: '',
  role: 1
};

export { memberDetails };
