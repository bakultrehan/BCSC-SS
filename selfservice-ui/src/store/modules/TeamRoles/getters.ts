import { GetterTree } from 'vuex';
import { TeamRoleState } from './types';
import { RootState } from '../../types';

/**
 * Getters Shared
 */
export const getters: GetterTree<TeamRoleState, RootState> = {
  /**
   * isLoading  status
   * @param {*} state
   * @returns {boolean} - loading status
   */
  isLoading(state) {
    return state.isLoading;
  },
  successStatus(state) {
    return state.successStatus;
  },
  errorStatus(state) {
    return state.errorStatus;
  },
  getTeamList(state) {
    return state.teamRoles;
  },
  memberErrorStatus(state) {
    return state.memberAddedError;
  },
  getMemberErrorList(state) {
    return state.memberAddedErrorList;
  },
  getMemberDetails(state) {
    return state.memberDetails;
  }
};
