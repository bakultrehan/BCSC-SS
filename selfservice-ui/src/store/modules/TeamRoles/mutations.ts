import { MutationTree } from 'vuex';
import { TeamRoleState } from './types';

/**
 * mutation
 */
export const mutations: MutationTree<TeamRoleState> = {
  /**
   * Sets loadinf
   * @param {*} state
   * @param {boolean} flag
   */

  SET_LOADING(state, flag: boolean) {
    state.isLoading = flag;
  },
  /**
   * set stae on success
   * @param {*} state
   * @param {*} payload
   */
  SET_ROLESLIST(state, payload: any) {
    state.teamRoles = payload;
  },

  /**
   * set stae on error
   * @param {*} state
   * @param {*} payload
   */
  SET_ERROR(state, payload: any) {
    state.errorStatus = payload;
  }
};
