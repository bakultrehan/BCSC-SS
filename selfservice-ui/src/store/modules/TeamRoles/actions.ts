import { ActionTree } from 'vuex';
import { TeamRoleState } from './types';
import { RootState } from '../../types';
import { TeamRoles } from '@/services/TeamRoles';

/**
 * TeamRoles Actions
 *
 */
export const actions: ActionTree<TeamRoleState, RootState> = {
  /**
   * load roles from server and set to store
   * @param {*} { commit }
   */
  async loadTeam({ commit }, projectId) {
    commit('SET_LOADING', true);
    try {
      const roles = await TeamRoles.getTeamRoles(projectId);
      commit('SET_ROLESLIST', roles.data.team);
      commit('SET_LOADING', false);
    } catch {
      commit('SET_ERROR', true);
      commit('SET_LOADING', false);
    }
  },
  /**
   * clear message
   * @param {*} { commit }
   */
  async clearStatus({ commit }) {
    commit('SET_SUCESS', false);
    commit('SET_ERROR', false);
    commit('SET_LOADING', false);
  },
  /**
   * add team member
   * @param {*} { commit }
   */
  async addTeamMember({ commit }, { userDetails, projectId }) {
    commit('SET_LOADING', true);
    try {
      const roles = await TeamRoles.addTeamMember(userDetails, projectId);
      commit('SET_MEMBER_ADDED', true);
      commit('SET_MEMBER_ADDED_ERROR', false);
      commit('SET_LOADING', false);
      commit('SET_MEMBER_ADDED_ERROR_LIST', {});
    } catch (error) {
      commit('SET_MEMBER_ADDED_ERROR_LIST', {});
      if (error.response.status === 400) {
        commit('SET_MEMBER_ADDED_ERROR_LIST', error.response.data.errors);
      }
      commit('SET_MEMBER_ADDED', false);
      commit('SET_MEMBER_ADDED_ERROR', true);
      commit('SET_LOADING', false);
    }
  },
  /**
   * get team member detail
   * @param {*} { commit }
   */
  async getTeamMember({ commit }, { projectId, memberId }) {
    commit('SET_LOADING', true);
    try {
      const memberDetails = await TeamRoles.getTeamMember(projectId, memberId);
      commit('SET_MEMBER_DETAILS', memberDetails.data);
      commit('SET_MEMBER_ADDED_ERROR', false);
      commit('SET_LOADING', false);
      commit('SET_MEMBER_ADDED_ERROR_LIST', {});
    } catch (error) {
      commit('SET_MEMBER_ADDED_ERROR_LIST', {});
      if (error && error.response && error.response.status === 400) {
        commit('SET_MEMBER_ADDED_ERROR_LIST', error.response.data.errors);
      }
      commit('SET_MEMBER_ADDED', false);
      commit('SET_MEMBER_ADDED_ERROR', true);
      commit('SET_LOADING', false);
    }
  },
  /**
   * update team member
   * @param {*} { commit }
   */
  async updateTeamMember({ commit }, { userDetails, projectId, memberId }) {
    commit('SET_LOADING', true);
    try {
      const roles = await TeamRoles.updateTeamMember(
        userDetails,
        projectId,
        memberId
      );
      commit('SET_MEMBER_ADDED', true);
      commit('SET_MEMBER_ADDED_ERROR', false);
      commit('SET_LOADING', false);
      commit('SET_MEMBER_ADDED_ERROR_LIST', {});
    } catch (error) {
      commit('SET_MEMBER_ADDED_ERROR_LIST', {});
      if (error && error.response && error.response.status === 400) {
        commit('SET_MEMBER_ADDED_ERROR_LIST', error.response.data.errors);
      }
      commit('SET_MEMBER_ADDED', false);
      commit('SET_MEMBER_ADDED_ERROR', true);
      commit('SET_LOADING', false);
    }
  }
};
