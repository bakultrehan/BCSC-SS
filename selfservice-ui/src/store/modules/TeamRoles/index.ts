import { Module } from 'vuex';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { TeamRoleState } from './types';
import { RootState } from '../../types';

/**
 * Sharedmodule state
 */
export const state: TeamRoleState = {
  teamRoles: {},
  isLoading: false,
  successStatus: false,
  errorStatus: false
};

const namespaced: boolean = true;

const SharedModule: Module<TeamRoleState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
};

export default SharedModule;
