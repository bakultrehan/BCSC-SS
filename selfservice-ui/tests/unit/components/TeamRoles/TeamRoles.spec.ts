import { shallowMount } from '@vue/test-utils';
import TeamRoles from '@/components/TeamRoles/TeamRoles.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('TeamRoles.vue', () => {
  const store = new Vuex.Store({
    modules: {
      TeamRolesModule: {
        namespaced: true,
        state: {},
        getters: {
          successStatus: jest.fn(),
          errorStatus: jest.fn(),
          isLoading: jest.fn(),

          getTeamList: jest.fn(() => {
            return [];
          })
        },
        actions: {
          loadTeam: jest.fn(),
          clearStatus: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return shallowMount(TeamRoles, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed with gettors', () => {
    const TeamRolesPage = mountFunction({});
    expect(TeamRolesPage.element).toMatchSnapshot();
  });
});
