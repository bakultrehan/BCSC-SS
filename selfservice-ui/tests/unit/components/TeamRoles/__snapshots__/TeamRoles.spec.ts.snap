// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamRoles.vue renders props when passed with gettors 1`] = `
<v-container-stub
  tag="div"
>
  <v-row-stub
    class="ma-5"
    tag="div"
  >
    <v-col-stub
      cols="12"
      tag="div"
    >
      <v-card-stub
        flat="true"
        loaderheight="4"
        tag="div"
      >
        <h2
          class="text-left tab-headline"
        >
          
          
        
        </h2>
         
        <h4
          class="text-left tab-headline font-weight-medium"
        >
          
          
        
        </h4>
         
        <!---->
         
        <v-card-actions-stub
          class="mt-5"
        >
          <v-spacer-stub />
           
          <button-stub
            color=""
            name="button"
            type="button"
          >
            
          </button-stub>
        </v-card-actions-stub>
      </v-card-stub>
    </v-col-stub>
     
    <v-col-stub
      tag="div"
    >
      <div
        class="text-center"
      >
        <v-dialog-stub
          class="text-left"
          closedelay="0"
          contentclass=""
          maxwidth="none"
          opendelay="0"
          origin="center center"
          retainfocus="true"
          transition="dialog-transition"
          width="71%"
        >
          <v-card-stub
            loaderheight="4"
            tag="div"
          >
            <addteammember-stub
              id="0"
              memberid="0"
            />
          </v-card-stub>
        </v-dialog-stub>
      </div>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;
